crates/modelardb_client/src/main.rs

```diff
@@ -214,56 +196,69 @@ fn repl(
 
 /// Execute an action, a command, or a query. Returns [`Error`] if the action, command, or query
 /// could not be executed or their result could not be retrieved.
-fn execute_and_print_action_command_or_query(
-    runtime: &Runtime,
+async fn execute_and_print_action_command_or_query(
     flight_service_client: &mut FlightServiceClient<Channel>,
     action_command_or_query: &str,
-) -> Result<(), Box<dyn Error>> {
-    if action_command_or_query.starts_with('\\') {
-        execute_command(runtime, flight_service_client, action_command_or_query)?;
-    } else if action_command_or_query.starts_with("SELECT")
+) {
+    let start_time = Instant::now();
+
+    let result = if action_command_or_query.starts_with('\\') {
+        execute_command(flight_service_client, action_command_or_query).await
+    } else if action_command_or_query.starts_with("INSERT")
         || action_command_or_query.starts_with("EXPLAIN")
+        || action_command_or_query.starts_with("SELECT")
     {
-        execute_query_and_print_result(runtime, flight_service_client, action_command_or_query)?;
+        execute_query_and_print_result(flight_service_client, action_command_or_query).await
     } else {
-        execute_action(runtime, flight_service_client, action_command_or_query)?;
+        execute_action(
+            flight_service_client,
+            "CommandStatementUpdate",
```

Comment:
Is it intentional that the only action that can be executed using this function is `CommandStatementUpdate`? If so, maybe specify this in the function doc comment.

Reply:
Other `Action`s are executed by `execute_action()` in `execute_command()` such as "FlushMemory" and "FlushEdge" so I am not entirely sure which function you mean that can only execute "CommandStatementUpdate".